docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -35,368 +35,1390 @@ a class to implement a particular interface. Serializers might be implemented an
 separately from the classes.
 
 It also supports zero-configuration use cases by automatically extracting schemas out of the
-classes and Java records using reflection, which is cached and reused later, with no extra cost.
+Java classes and records, and .NET classes which are cached and reused later, with no extra cost.
 
 The underlying format of the Compact serialized objects is platform and language independent.
 
-== Using Compact Serialization With Zero-Configuration
-
-Compact serialization can be used without a registering the serializer in the member or client
-configuration. Hazelcast tries
-to extract a schema out of the class, using reflection. If successful, it registers the
-reflective serializer associated with the extracted schema and uses it while
-serializing and deserializing instances of that class. If the automatic schema
-extraction fails, Hazelcast throws an exception.
+NOTE: Compact serialization is promoted to the stable status in the 5.2 Hazelcast release.
+The older versions released under the BETA status are not compatible with stable the 5.2 version. +
+For future versions, the backward compatibility will be maintained, just like any other
+Hazelcast feature.
 
```

Comment:
@mdumandag , don't you think this a release notes or migration guide material? (because our product documentation are versioned)

Reply:
Maybe we can reword the related entry at https://docs.hazelcast.com/hazelcast/5.2/release-notes/5-2-0#breaking-changes

Reply:
Yes, we can put this information in a couple of places, but my impression was that since we were beta in 5.1 and 5.0 for compact, and only promoted to the stable status in 5.2, I don't think it is a breaking change. We never gave a guarantee of future compatibility for compact beta

Reply:
But, this is definitely important information, so maybe we can put it in some place in the release notes